rust/gelato/src/chains.rs

```diff
@@ -0,0 +1,151 @@
+// Ideally we would avoid duplicating ethers::types::Chain, but we have enough need to justify
+// a separate, more complete type conversion setup, including some helpers for e.g. locating
+// Gelato's verifying contracts. Converting from the chain's name in string format is useful
+// for CLI usage so that we don't have to remember chain IDs and can instead refer to names.
+
+use ethers::types::{Address, U256};
+use std::{fmt, str::FromStr};
+
+use crate::err::GelatoError;
+
+// This list is currently trimmed to the *intersection* of
+// {chains used by Abacus in any environment} and {chains included in ethers::types::Chain}.
+// Notably missing is Celo/Alfajores.
+#[derive(Debug, Clone, Copy, Eq, PartialEq)]
+pub enum Chain {
```

Comment:
If I were you I would consider using a one-off (private) macro to prevent needing to list the same values more than once. I can help if you are interested.

Reply:
I have mostly modeled this after the ethers::types::Chain implementation here: https://github.com/gakonst/ethers-rs/blob/master/ethers-core/src/types/chain.rs, which does not use a macro. I don't personally mind listing things twice, if you're ok with it, could we skip the macro for now?